### PR TITLE
CLOUDP-140309: Test AtlasCLI repo is failing during the release:

### PR DESCRIPTION
**Pull request by andreaangiolillo**
## Proposed changes

_Jira ticket:_ CLOUDP-140309

## Checklist

- [x] I have signed the [MongoDB CLA](https://www.mongodb.com/legal/contributor-agreement)
- [ ] I have added tests that prove my fix is effective or that my feature works
- [x] I have added any necessary documentation in document requirements section listed in [CONTRIBUTING.md](https://github.com/mongodb/mongodb-atlas-cli/blob/master/CONTRIBUTING.md) (if appropriate)
- [x] I have addressed the @mongodb/docs-cloud-team comments (if appropriate)
- [x] I have updated [test/README.md](https://github.com/mongodb/mongodb-atlas-cli/blob/master/test/README.md) (if an e2e test has been added)
- [x] I have run `make fmt` and formatted my code

## Further comments

The task [Test AtlasCLI repo](https://evergreen.mongodb.com/task/mongocli_master_release_atlascli_repo_test_pkg_test_repo_atlascli__atlascli_v1.2.1_22_10_13_10_15_48) failed during the release. The issue seems cause by the publish task renaming happened in https://github.com/mongodb/mongodb-atlas-cli/pull/1448

### CODE REVIEW

File: tools/genevergreen/generate/generate.go [hunk 1]
```diff
@@ -191,7 +199,7 @@ func PublishStableTasks(c *shrub.Configuration, toolName string) {
 	}
 	for _, sv := range serverVersions {
 		v := &shrub.Variant{
-			BuildName:        fmt.Sprintf("generated_release_%s_publish_%s", toolName, sv),
+			BuildName:        fmt.Sprintf("generated_release_%s_publish_%s", toolName, strings.ReplaceAll(sv, ".", "")),
```
Comment (andreaangiolillo) [inline tools/genevergreen/generate/generate.go:202]: removing the `.` from the variant name. From `test_repo_atlascli_amazonlinux2_org_4.2 `to `test_repo_atlascli_amazonlinux2_org_42 ` as it was before https://github.com/mongodb/mongodb-atlas-cli/pull/1448

File: tools/genevergreen/generate/generate.go [hunk 2]
```diff
@@ -70,11 +70,19 @@ var (
...
+func newDependency(toolName, os, serverVersion, repo string) []shrub.TaskDependency {
+	var deps []shrub.TaskDependency
+	architectures := Distros[newOs[os]].architectures
+
+	for _, arch := range architectures {
+		dep := shrub.TaskDependency{
+			Name:    fmt.Sprintf("push_%s_%s_%s_%s_%s_stable", toolName, newOs[os], repo, arch, strings.ReplaceAll(serverVersion, ".", "")),
+			Variant: fmt.Sprintf("generated_release_%s_publish_%s", toolName, strings.ReplaceAll(serverVersion, ".", "")),
+		}
+		deps = append(deps, dep)
```
Comment (andreaangiolillo) [inline tools/genevergreen/generate/generate.go:82]: from `push_atlascli_amazon2_enterprise_stable` to `push_atlascli_amazon2_org_aarch64_42_stable`

**Review by fmenezes — APPROVED** [review]
LGTM

File: tools/genevergreen/generate/generate.go [hunk 3]
```diff
@@ -70,11 +70,19 @@ var (
...
 
-func newDependency(toolName, os, serverVersion, repo string) shrub.TaskDependency {
-	return shrub.TaskDependency{
-		Name:    fmt.Sprintf("push_%s_%s_%s_stable", toolName, newOs[os], repo),
-		Variant: fmt.Sprintf("release_%s_publish_%s", toolName, strings.ReplaceAll(serverVersion, ".", "")),
+func newDependency(toolName, os, serverVersion, repo string) []shrub.TaskDependency {
+	var deps []shrub.TaskDependency
+	architectures := Distros[newOs[os]].architectures
+
+	for _, arch := range architectures {
```
Comment (gssbzn) [inline tools/genevergreen/generate/generate.go:77]: this test is currently written to test amd64 version of the bins, depending on the ARM publishing won't do much for these test

Comment (andreaangiolillo) [reply tools/genevergreen/generate/generate.go]: updated

**Review by gssbzn — APPROVED** [review]
LGTM

**Pull request merged**